### DISCUSSION TITLE
Handle configuration files that are externally modified and also in different locations on difference machines

### DISCUSSION THREAD

**Discussion by Tyrubias** [twpayne/chezmoi#1312] [Q&A] [answered]
The `chezmoi` docs has sections on handling externally modified config files as well as config files that exist in different locations on difference systems. However, I'm not sure how to combine the info to handle externally modified config files that exist in difference locations on different systems.

For context, the file I have in mind is VSCode's `settings.json`, which exists in different location on macOS and Linux. Any pointers for doing this?

**Comment by nandalopes** [comment]
I think you may handle this within .chezmoiignore file: if not Linux, ignore linux location of files; and so on

**Comment by twpayne** [comment, marked as answer]
In the particular case of VSCode, VSCode has its own [system for synchronizing settings](https://code.visualstudio.com/docs/editor/settings-sync). Depending on exactly what you want, this might be the best option for you.

For managing `settings.json` on chezmoi you could use a per OS-symlink back to the source state, something like:

`~/.local/share/chezmoi/settings.json`:
```
{ /* ... */ }
```

`.local/share/chezmoi/.chezmoiignore`:
```
settings.json
{{ if ne .chezmoi.os "darwin" }}
Library/Application Support/Code/User/settings.json
{{ end }}
{[ if ne .chezmoi.os "linux" }}
./.config/Code/User/settings.json
{{ end }}
```

`.local/share/chezmoi/Library/Application Support/Code/User/symlink_settings.json.tmpl`:
```
{{ .chezmoi.sourceDir }}/settings.json
```

`.config/Code/User/symlink_settings.json.tmpl`:
```
{{ .chezmoi.sourceDir }}/settings.json
```

**Reply by lnyousif** [reply]
@twpayne I know this is an old answer but I am failing t understand the .chezmoiignore , did you mean this 

```
settings.json
{{ if ne .chezmoi.os "darwin" }}
Application Support/Code/User/symlink_settings.json.tmpl
{{ end }}
{{ if ne .chezmoi.os "linux" }}
.config/Code/User/symlink_settings.json.tmpl
{{ end }}
```

**Reply by Lockszmith-GH** [reply]
The paths in `.chezmoiignore` are from the source and not from the chezmoi cache.

Copy the text verbatime from **twpayne**'s example.

Did you try that and it failed you? (it might benefit you to just open a new discussion about your particular issue)

**Reply by lnyousif** [reply]
No I was trying to understand the concept behind it, thanks for explaining , 

copying the code would not work , there is {[ instead of {{ but i guess the rest should be fine per your explanation

**Reply by halostatue** [reply]
The paths in `.chezmoiignore` are from the target value, so any chezmoi-specific attributes must be removed:

```
{{ if ne .chezmoi.os "darwin" }}
Library/Application Support/Code/User/settings.json
{{ end }}
{{ if ne .chezmoi.os "linux" }}
.config/Code/User/settings.json
{{ end }}
```

You can't have your shared settings value in `.chemzoiignore` since that will prevent it from being processed at all.

What you probably want to do is something closer to having the Linux location be your source (`private_dot_config/code/user/settings.json.tmpl`) and then have `private_Library/Application Support/Code/User/symlink_settings.json.tmpl`:

```
{{- if .chemzoi.os "darwin" }}{{ joinPath .chezmoi.home ".config/code/user/settings.json" }}{{ end -}}
```

Then you don't need to have anything in `.chezmoiignore` for this, because an empty symlink file will not be applied.

**Reply by lnyousif** [reply]
Thank you, this is very helpful
